### PR TITLE
Specify dependencies in stopping/deleting docker services

### LINKED ISSUE #1373: when codalab_service.py delete, stop monitor script first to avoid error messages
In general, add `depends_on` in the docker compose file so services are stopped in the right order.  Don't have to worry about starting, since `codalab_service.py` handles that manually.

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #1373

### CODE REVIEW

**Comment by candicegjing** [comment]
I didn't see any error in my local machine without specifying all those depends_on items. However, based on docker documentation, specifying depends_on will make sure the order of execution.

File: docker/compose_files/docker-compose.yml [hunk 1]
```diff
@@ -106,6 +106,12 @@ services:
     <<: *codalab-base
     depends_on:
       - rest-server
+      - bundle-manager
+      - frontend
```
Comment (percyliang) [inline docker/compose_files/docker-compose.yml]: Doesn't depend on frontend

File: docker/compose_files/docker-compose.yml [hunk 2]
```diff
@@ -106,6 +106,12 @@ services:
     <<: *codalab-base
     depends_on:
       - rest-server
+      - bundle-manager
```
Comment (percyliang) [inline docker/compose_files/docker-compose.yml]: Doesn't depend on bundle-manager

File: docker/compose_files/docker-compose.yml [hunk 3]
```diff
@@ -106,6 +106,12 @@ services:
...
+      - frontend
+      - nginx
+      - mysql
+      - worker
```
Comment (percyliang) [inline docker/compose_files/docker-compose.yml]: Doesn't depend on worker

File: docker/compose_files/docker-compose.yml [hunk 4]
```diff
@@ -106,6 +106,12 @@ services:
     <<: *codalab-base
     depends_on:
       - rest-server
+      - bundle-manager
+      - frontend
+      - nginx
+      - mysql
```
Comment (percyliang) [inline docker/compose_files/docker-compose.yml]: mysql is technically a dependency, but mysql should be a dependency of rest-server and bundle-manager

**Review by percyliang — CHANGES_REQUESTED** [review]
Might as well and go through each service and write down exactly what services it directly depends on.

File: docker/compose_files/docker-compose.yml (hunk 4 above)
Comment (candicegjing) [reply docker/compose_files/docker-compose.yml]: I think I misunderstood the meaning of *stopping monitor service first*. I have updated the code and split all the dependencies for each service.

**Review by percyliang — APPROVED** [review]

**Pull request merged**